### PR TITLE
FJS | Calculated Data in DataSource in Edit Mode is not working

### PR DESCRIPTION
**Pull request by yavorovsky**

### CODE REVIEW

**Review by irshadmb — APPROVED** [review]

**Comment by travist** [comment]
Please write some tests.

**Comment by yavorovsky** [comment]
@travist, done

**Comment by yavorovsky** [comment]
@travist , I updated the renders and merged master to pass tests.

File: src/components/_classes/component/Component.js [hunk 1]
```diff
@@ -2604,13 +2604,17 @@ export default class Component extends Element {
...
+    let calculatedValue = dataValue;
     // Calculate the new value.
-    let calculatedValue = this.evaluate(this.component.calculateValue, {
-      value: dataValue,
-      data,
-      row: row || this.data
-    }, 'value');
+
+    // If data was calculated in a submission and the editing mode is on, skip calculating
+    if (!flags.fromSubmission && !this.root.editing || !this.component.persistent) {
```
Comment (travist) [inline src/components/_classes/component/Component.js:2611]: These checks should just be added to the checks on line 2596.

**Review by travist — CHANGES_REQUESTED** [review]

File: src/components/_classes/component/Component.js (hunk 1 above)
Comment (travist) [reply src/components/_classes/component/Component.js:2611]: Disregard this comment.

**Pull request merged**